ls1intum/Artemis
# `Discussion`: Prefix post title with context and highlight announcements

### Checklist
#### General


#### Client

### Motivation and Context


Some change requests of instructors, tutors

### Description


**Changes include:**
1. Add title prefix (only in course discussion view, where posts of multiple contexts are listed)
- context name is used (Lecture title (with link to page), Exercise title (with link to page), Course-Wide topic (without link)) as prefix before the title
- (linked) context name in removed the post footer

2. Improvements for Announcements:
- content is highlighted
- no similarity check during creation

3. Markdown Fix:
- linebreaks in the post context are currently not rendered. With the changes in this PR, this is fixed by adding a lightweight custom pipe.

### Steps for Testing


1. Log in as instructor (for announcement in step 8) to Artemis
2. Navigate to course discussion overview
3. Check if the context is added as prefix at the beginning of the title, also if a post is resolved/ is the post is an announcement
4. Navigate to an exercise page
5. Check if the title is _not_ prefixed
6. Navigate to a lecture page
7. Check is the title is _not_ prefixed
8. Create an announcement by clicking on New post in the discussion overview
9. During entering a title, no similar posts should be shown (check disabled for this page)
11. In the post content add a linebreak, try with and without any #ID pattern
10. On saving the announcement, it should be shown, the content should be highlighted as seen in the screencast
11. The linebreak should be shown properly

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


**Before (context in footer):**
![image](https://user-images.githubusercontent.com/41366522/139494928-ea6ed54c-61bf-4e3f-8c55-72cde48e6913.png)


**After: Post Title for announcement, resolved post in lecture, post in exercise**
<img width="1038" alt="Bildschirmfoto 2021-10-29 um 13 06 41" src="https://user-images.githubusercontent.com/41366522/139493035-b4371b76-a83e-49e3-a17d-3544ccffc0e1.png">
